### ISSUE TITLE
function invocation

### ISSUE THREAD

**Issue by Bab64** [al66/imicros-feel-interpreter#1]
Considering this FEEL expression, say `exp`: `if count(X) > 0 then sum(for i in X return My Function(i)) else 0`, is there a way within the API to set up and next pass the `My Function` function through `data`: `interpreter.evaluate(exp,data);`
The following code of course doesn't work since `My Function` is missing: `interpreter.evaluate("if count(X) > 0 then sum(for i in X return My Function(i)) else 0",{"X": [{"value": 2},{"value": 1}]});`

To sum up, how to inject `My Function`, whose JS body could be: `function My Function(p) {return p.value * 2/3;}`?

Thansk in advance.

**Comment by al66** [comment]
Due to security reasons you cannot inject functions from outside. 
But related to your exmple you can define a function in a boxed expression or context and use it afterwords in another part.
There is an example in the tests with the defintion of the function calc(a,b):
`{{calc:function (a:number,b:number) a-b, y:calc(b:c,a:d)+3}.y`
The delivered value is only the calculated variable y. 
Here is the full example called with a context:
<code js>let result = interpreter.evaluate("{calc:function (a:number,b:number) a-b, y:calc(b:c,a:d)+3}.y",{c:4,d:5});
expect(result).toEqual(4);
</code>
